gradle/scans-init.gradle.kts

```diff
@@ -0,0 +1,47 @@
+import com.gradle.enterprise.gradleplugin.GradleEnterprisePlugin
+import com.gradle.scan.plugin.BuildScanPlugin
+import java.io.ByteArrayOutputStream
+
+/**
+ * To publish scan within the build, run
+ * ./gradlew build -I gradle/scans-init.gradle.kts
+ */
+initscript {
```

Comment:
What is the benefit of the init.gradle script instead of applying this to `settings.gradle` with a `publishAlwaysIf { }` block?

Reply:
1. I was not aware of `publishAlwaysIf` option
1. I followed instructions from gradle https://docs.gradle.com/enterprise/gradle-plugin/#scans_gradle_com

Does it have any performance impact? Is it considered as bad practice to implement it that way?

Reply:
Sorry, it sounds bad :) I am just curious :)

Reply:
Do you think it will be better to use, for instance `-PpublishScan` and use `publishAlwaysIf { [publishScan is present] }` ?

Reply:
I would recommend the instructions at the top of the page since you are only integrating the plugin with a single project. https://docs.gradle.com/enterprise/gradle-plugin/#applying_the_plugin

There is no significant performance impact. [Very large software companies](https://gradle.com/customers/) with very large Gradle builds use this plugin for all their builds. Build scans are uploaded in the background.

Reply:
Thanks, refactored